packages/ra-core/src/controller/create/useCreateController.ts

```diff
@@ -91,55 +95,67 @@ export const useCreateController = <
-                                                'ra.notification.http_error',
-                                      {
-                                          type: 'warning',
-                                          messageArgs: {
-                                              _:
-                                                  typeof error === 'string'
-                                                      ? error
-                                                      : error && error.message
-                                                      ? error.message
-                                                      : undefined,
-                                          },
-                                      }
-                                  );
-                              },
+                                notify('ra.notification.created', {
+                                    type: 'info',
+                                    messageArgs: { smart_count: 1 },
+                                });
+                                redirect(
+                                    finalRedirectTo,
+                                    resource,
+                                    data.id,
+                                    data
+                                );
+                            },
+                            onError: onErrorFromSave
+                                ? onErrorFromSave
+                                : onError
+                                ? onError
+                                : (error: Error) => {
+                                      notify(
+                                          typeof error === 'string'
+                                              ? error
+                                              : error.message ||
+                                                    'ra.notification.http_error',
+                                          {
+                                              type: 'warning',
+                                              messageArgs: {
+                                                  _:
+                                                      typeof error === 'string'
+                                                          ? error
+                                                          : error &&
+                                                            error.message
+                                                          ? error.message
+                                                          : undefined,
+                                              },
+                                          }
+                                      );
+                                  },
+                        }
+                    );
+                } catch (error) {
+                    if ((error as HttpError).body?.errors != null) {
+                        return (error as HttpError).body.errors;
```

Comment:
This is weird: the save promise *resolves* when there is a validation error in the save process? I'd expect it to reject.

Reply:
This is how the `useAugmentedForm` has been implemented. If the submit function returns something, we consider this is an error object. This was done like this probably because it worked that way in `final-form`.

Reply:
can't we improve that in a backward compatible way?